examples/tutorials/self-paced-training/part-3_security_and_privacy/chapter-6_Security_in_federated_compute_system/06.4_customized_site_security/custom_server_side_authentication/server_side_security_plugin.ipynb

```diff
@@ -0,0 +1,163 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "c1f37174",
+   "metadata": {},
+   "source": [
+    "# Server Side Customized Authenticaton\n",
+    "\n",
+    "In this example, we will see how do we designed a custom plugin with additional authentication check, \n",
+    "As result for two sites in POC, site-2 is NOT able to start and register to the server. It's blocked by the ServerCustomSecurityHandler logic during the client registration.\n",
+    "\n",
+    "## Define a server side security handler\n",
+    "\n",
+    "Notice the we the customized the handler raise NotAuthenticated(\"site_2 not allowed to register\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "235ed0fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat security/server/custom/security_handler.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58c8f45a",
+   "metadata": {},
+   "source": [
+    "To register this plugin handler, we need to add this component to the server site's local configuration\n",
+    "\n",
+    "by adding it to the components array \n",
+    "\n",
+    "```\n",
+    "    components: [\n",
+    "    ...\n",
+    "            {\n",
+    "                \"id\": \"security_handler\",\n",
+    "                \"path\": \"security_handler.ServerCustomSecurityHandler\"\n",
+    "            }\n",
+    "    ]  \n",
+    "```  \n",
+    "\n",
+    "In this example, we will copy \"custom\" folder and \n",
+    "\n",
+    "use python code ```edit_site_local_resources.py``` to create \"resources.json\" to the at site local directory\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d5bfe22f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "! echo y | nvflare poc prepare"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4c63cb5a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "\n",
+    "!cp -r security/server/* /tmp/nvflare/poc/example_project/prod_00/server/local/.\n",
+    "!python edit_site_local_resources.py server /tmp/nvflare/poc/example_project/prod_00"
```

Comment:
I think it is clearer to just create a "resources.json" that contains this security_handler.
then in here we do "cp" instead of running this python script.